serif_cli/src/main/kotlin/serif_cli/App.kt

```diff
@@ -31,43 +33,46 @@ class App {
             return
         }
 
-        //Each UI will create it's specific DriverFactory
-        //And call this function before the backend can get
-        //information out of the database
+        // Each UI will create it's specific DriverFactory
+        // And call this function before the backend can get
+        // information out of the database
         Database.initDb(DriverFactory())
 
+        val getInputToQueue: () -> Unit = { thread(start = true) { queue.add(console.readLine()) }; }
```

Comment:
Do we need to remake this thread in each new state? Can't we run this at the top level and use it in place of readLine() calls in the login?

Reply:
Perhaps, but at least our readPassword call has different behavior. This issue is probably obsoleted by your pull request anyway, assuming we can get lanterna to play nice!